devdoc/doublylinkedlist_requirements.md

```diff
@@ -104,3 +127,54 @@ extern PDLIST_ENTRY DList_RemoveHeadList(PDLIST_ENTRY listHead);
 **SRS_DLIST_06_013: [** DList_RemoveHeadList shall return listHead if that's the only item in the list. **]**
 
 Note: The Flink & Blink of the returned PDLIST_ENTRY shall be undefined.
+
+### DList_Find
+```c
+MOCKABLE_FUNCTION(, int, DList_Find, PDLIST_ENTRY, listHead, DLIST_MATCH_FUNCTION, matchFunction, const void*, matchContext, PDLIST_ENTRY*, foundEntry);
+```
+
+**SRS_DLIST_43_001: [** `DList_Find` shall call `matchFunction` on each entry in the list defined by `listHead` along with `matchContext`. **]**
+
+**SRS_DLIST_43_002: [** If the call to `matchFunction` for an entry returns `DLIST_MATCH_FUNCTION_MATCHING`, `DList_Find` shall set `foundEntry` to that entry. **]**
+
+**SRS_DLIST_43_003: [** If no calls to `matchFunction` return `DLIST_MATCH_FUNCTION_MATCHING`, `DList_Find` shall set `foundEntry` to `NULL`. **]**
+
+**SRS_DLIST_43_015: [** `DList_Find` shall succeed and return zero. **]**
```

Comment:
Why not have an enum return?
FOUND/NOT_FOUND/ERROR?
That would allow checking it in a switch statement.
The way it is now one has to first check the int return and then as a second check one has to check whether foundEntry is NULL or not. 

I mean, both work, but my personal preference would be to have an enum as return.

Reply:
Changing to enum